src/python/platforms/android/flash.py

```diff
@@ -51,6 +55,36 @@
 FLASH_REBOOT_BOOTLOADER_WAIT = 15
 FLASH_REBOOT_WAIT = 5 * 60
 
+def download_latest_build(branch, target, image_regexes):
+  """Download the latest build artifact for the given branch and target."""
+  build_info = fetch_artifact.get_latest_artifact_info(branch, target)
+  if not build_info:
+    logs.log_error(
+        'Unable to fetch information on latest build artifact for '
+        'branch %s and target %s.' % (branch, target))
+    return
+  # Check if our local build matches the latest build. If not, we will
+  # download it.
+  build_id = build_info['bid']
+  target = build_info['target']
+  image_directory = environment.get_value('IMAGES_DIR')
+  last_build_info = persistent_cache.get_value(
+      constants.LAST_FLASH_BUILD_KEY)
+  if not last_build_info or last_build_info['bid'] != build_id:
+    # Clean up the images directory first.
+    shell.remove_directory(image_directory, recreate=True)
+
+    for image_regex in image_regexes:
+      image_file_path = fetch_artifact.get(build_id, target, image_regex,
+                                            image_directory)
+      if not image_file_path:
+        logs.log_error(
+            'Failed to download artifact %s for '
+            'branch %s and target %s.' % (image_file_path, branch, target))
+        return
+      if image_file_path.endswith('.zip'):
```

Comment:
why don't we untar here as well instead of in adb.py ?

Reply:
I wanted to make the download and copy faster, but agree if we untar here the code is cleaner.